Alignment/OfflineValidation/bin/Zmumumerge.cc

```diff
@@ -0,0 +1,310 @@
+      if (i == 7 and j > 25) {
+        continue;
+      }
+      cout << "th1d_variables_meanmass[i]->GetNbinsX()=" << th1d_variables_meanmass[i]->GetNbinsX() << endl;
+      cout << "th2d_mass_variables[i]->GetNbinsY()=" << th2d_mass_variables[i]->GetNbinsY() << endl;
+      th1d_variables_meanmass[i]->SetBinContent(j, 0);
+      th1d_variables_meanmass[i]->SetBinError(j, 0);
+
+      TString th1d_mass_temp_name = Form("th1d_mass_%s_%d", tstring_variables_name[i].Data(), j);
+      TH1D* th1d_i = th2d_mass_variables[i]->ProjectionX(th1d_mass_temp_name, j, j, "d");
+      th1d_i->Write(th1d_mass_temp_name);
+      TString s_cut = Form("nocut");
+      TString s_name = Form("%s_%d", tstring_variables_name[i].Data(), j);
+
+      FitOut fitR = ZMassBinFit_OldTool(th1d_i, s_name, output_path);
+      th1d_variables_meanmass[i]->SetBinContent(j, fitR.mean);
+      th1d_variables_meanmass[i]->SetBinError(j, fitR.mean_err);
+    }
+    th1d_variables_meanmass[i]->GetXaxis()->SetRangeUser(xaxis_range[i][0], xaxis_range[i][1]);
+    Draw_th1d(th1d_variables_meanmass[i], tstring_variables_name[i]);
+    outpufile->cd();
+    th1d_variables_meanmass[i]->Write(th1d_name);
+
+    TString th1d_name_entries = Form("th1d_entries_%s", tstring_variables_name[i].Data());
+    th1d_variables_entries[i] = th2d_mass_variables[i]->ProjectionY(th1d_name_entries, 0, -1, "d");
+    th1d_variables_entries[i]->GetXaxis()->SetTitle(tstring_variables_name[i].Data());
+    th1d_variables_entries[i]->GetYaxis()->SetTitle("Entry");
+    outpufile->cd();
+    th1d_variables_entries[i]->Write(th1d_name_entries);
+  }
+
+  outpufile->Write();
+  outpufile->Close();
+  delete outpufile;
+}
+
+const static int max_file_number = 10;
+void Draw_TH1D_forMultiRootFiles(
+    int file_number, vector<TString> file_names, vector<TString> label_names, TString th1d_name, TString output_name) {
+  TH1D* th1d_input[max_file_number];
+  TFile* file_input[max_file_number];
+  for (int idx_file = 0; idx_file < file_number; idx_file++) {
+    file_input[idx_file] = TFile::Open(file_names[idx_file]);
+    th1d_input[idx_file] = (TH1D*)file_input[idx_file]->Get(th1d_name);
+  }
+
+  TCanvas* c = new TCanvas();
+  TLegend* lg = new TLegend(0.2, 0.7, 0.5, 0.95);
+  c->cd();
+  gStyle->SetOptStat(0);
+  th1d_input[0]->SetTitle("");
+  for (int idx_file = 0; idx_file < file_number; idx_file++) {
+    th1d_input[idx_file]->SetMarkerColor(colors_array[idx_file]);
+    th1d_input[idx_file]->SetLineColor(colors_array[idx_file]);
+    th1d_input[idx_file]->Draw("same");
+    lg->AddEntry(th1d_input[idx_file], label_names[idx_file]);
+  }
+  lg->Draw("same");
+  c->SaveAs(output_name);
+}
+
+int Zmumumerge(int argc, char* argv[]) {
+  vector<TString> vec_single_file_path;
+  vector<TString> vec_single_file_name;
+  vector<TString> vec_global_tag;
+  vec_single_file_path.clear();
+  vec_single_file_name.clear();
+  vec_global_tag.clear();

```

Comment:
Why are you using `clear` right after declaring the vectors? Is there a particular reason?

Naïvely, I'd say that these lines aren't necessary, but I might be missing sth.

Reply:
I have deleted these codes.